solidity/contracts/token/ValueTransferBridgeNative.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+import {HypNative} from "./HypNative.sol";
+import {TypeCasts} from "../libs/TypeCasts.sol";
+import {TokenMessage} from "./libs/TokenMessage.sol";
+import {TokenRouter} from "./libs/TokenRouter.sol";
+import {IValueTransferBridge} from "../interfaces/IValueTransferBridge.sol";
+import {StandardHookMetadata} from "../hooks/libs/StandardHookMetadata.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+/**
+ * @title Native/ERC20 tokens L2 to L1 value transfer abstraction
+ * @author Substance Labs
+ * @dev Derives from the Hyperlane native token router, but supports
+ * transfer of ERC20 token value
+ */
+abstract contract ValueTransferBridgeNative is HypNative {
+    using TypeCasts for bytes32;
+    using TypeCasts for address;
+    using TokenMessage for bytes;
+
+    uint32 constant HOOK_METADATA_GAS_LIMIT = 450_000;
+
+    // L2 bridge used to initiate the withdrawal
+    address public l2Bridge; // TODO: immutable
+    // L1 domain where the withdrawal will be finalized
+    uint32 public immutable l1Domain;
+
+    error NotImplemented();
+
+    /**
+     * @dev see MailboxClient's initializer for other configurables
+     */
+    constructor(
+        uint32 _l1Domain,
+        address _l2Bridge,
+        address _mailbox
+    ) HypNative(_mailbox) {
+        l1Domain = _l1Domain;
+        l2Bridge = _l2Bridge;
+        _transferOwnership(_msgSender()); // TODO: remove
+    }
+
+    function setL2Bridge(address _l2Bridge) external onlyOwner {
+        l2Bridge = _l2Bridge;
+    }
+
+    function quoteTransferRemote(
```

Comment:
This method should return `Quote[]` so that consumers of the `IValueTransferBridge` contracts can support bridges where the quote is in the token being transferred (OpStack bridge admittedly is not one of them)